mantidproject/mantid
# Add support for a Python 3 build on RHEL 7

**Description of work.**

Contains packaging and other fixes when compiling against python 3 on RHEL 7. Note that this does not enable Python 3 by default yet. It includes building a private copy of sip/PyQt4 in Python 3 as the sip available from EPEL causes PyQt4 to segfault.

**To test:**

It currently requires the standard mantid/mantid copr and [my sandbox](https://copr.fedorainfracloud.org/coprs/martyngigg/sandbox/). Install these with

```
yum install yum-plugin-copr
yum copr enable mantid/mantid
yum copr enable martyngigg/sandbox
```

Build with cmake and `-DPYTHON_EXECUTABLE=/usr/bin/python3`. Check MantidPlot and workbench.

Fixes #27631

*This does not require release notes* because **Python 3 migration will be announced separately**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
